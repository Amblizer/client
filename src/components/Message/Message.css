/* MESSANGER */

.message-operatorPicture-messenger {
  width: 48px;
  height: 48px;
  float: right;
  margin-top: 4px;
  box-sizing: border-box;
  padding: 0 4px;
}

.message-operatorPictureImage-messenger {
  width: 40px;
  height: 40px;
}

.message-operatorContent-messenger {
  margin: 0;
  padding: 6px;
  list-style: none;
  width: 160px;
  margin-top: 4px;
  margin-bottom: 4px;
  border-radius: 10px 0 10px 10px;
  background: #e1e1e1;
  color: #000000;
  float: right;
  text-align: right;
}

.message-userContent-messenger {
  margin: 0;
  list-style: none;
  width: 160px;
  float: left;
  text-align: left;
  background: #0a6bef;
  margin-top: 0;
  margin-left: 4px;
  border-radius: 10px 10px 10px 0;
  padding: 6px;
  color: #ffffff;
}

/* .letschat-message { } */
/* .letschat-message-timestamp{ }*/

/* FLOAT */

.message-operatorPicture-float {
  position: absolute;
  left: -48px;
  width: 48px;
  height: 48px;
  float: right;
  margin-top: -16px;
  box-sizing: border-box;
  padding: 0 4px;
}

.message-operatorPictureImage-float {
  width: 40px;
  height: 40px;
  border-radius: 20px;
}

.message-operatorContent-float {
  margin: 0;
  list-style: none;
  width: 160px;
  float: left;
  text-align: left;
  background: #ffffff;
  color: #000000;
  margin-top: 4px;
  margin-left: 4px;
  box-shadow: #dddddd 1px 1px 8px 0;
  border-radius: 0 10px 10px 10px;
  padding: 16px;
}

.message-userContent-float {
  margin: 0;
  padding: 16px;
  list-style: none;
  width: 160px;
  margin-top: 4px;
  margin-bottom: 4px;
  box-shadow: #dddddd 1px 1px 8px 0;
  border-radius: 10px 10px 0 10px;
  background: #0a6bef;
  float: right;
  text-align: right;
  color: #ffffff;
  }


/* SIDEPANEL */

.message-operatorPicture-sidepanel {
  width: 48px;
  height: 48px;
  float: right;
  margin-top: 4px;
  box-sizing: border-box;
  padding: 0 4px;
}

.message-operatorPictureImage-sidepanel {
  width: 40px;
  height: 40px;
}

.message-operatorContent-sidepanel {
  margin: 0;
  padding: 6px;
  list-style: none;
  width: 160px;
  margin-top: 4px;
  margin-bottom: 4px;
  border-radius: 10px 0 10px 10px;
  background: #e1e1e1;
  color: #000000;
  float: right;
  text-align: right;
}

.message-userContent-sidepanel {
  margin: 0;
  list-style: none;
  width: 160px;
  float: left;
  text-align: left;
  background: #0a6bef;
  margin-top: 0;
  margin-left: 4px;
  border-radius: 10px 10px 10px 0;
  padding: 6px;
  color: #ffffff;
}