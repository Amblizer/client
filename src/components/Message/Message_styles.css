.Message-userContent {
  margin: 0;
  list-style: none;
  width: 160px;
  float: left;
  text-align: left;
  background: #0a6bef;
  margin-top: 0;
  margin-left: 4px;
  border-radius: 10px 10px 10px 0;
  padding: 6px;
  color: white;
}

.Message-operatorPictureImage{
  width: 40px;
  height: 40px;
}

.Message-operatorPictureImage{
  width: 40px;
  height: 40px;
}

.Message-operatorPicture {
  width: 48px;
  height: 48px;
  float: right;
  marginTop: 4px;
  boxSizing: border-box;
  padding: 0 4px;
}

.Message-operatorContent {
  margin: 0;
  padding: 6px;
  list-style: none;
  width: 160px;
  margin-top: 4px;
  margin-bottom: 4px;
  border-radius: 10px 0 10px 10px;
  background: #e1e1e1;
  float: right;
  text-align: right;
}

.letschat-message {

}

.letschat-message-timestamp{

}