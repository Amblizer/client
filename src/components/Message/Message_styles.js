const styles = {
  Message: {
    operatorPicture: {
      width: '48px',
      height: '48px',
      float: 'right',
      marginTop: '4px',
      boxSizing: 'border-box',
      padding: '0 4px',
    },
    operatorPictureImage: {
      width: '40px',
      height: '40px',
    },
    operatorContent: {
      margin: 0,
      padding: '6px',
      listStyle: 'none',
      width: '160px',
      marginTop: '4px',
      marginBottom: '4px',
      borderRadius: '10px 0 10px 10px',
      background: '#e1e1e1',
      float: 'right',
      textAlign: 'right',
    },
    userContent: {
      margin: 0,
      listStyle: 'none',
      width: '160px',
      float: 'left',
      textAlign: 'left',
      background: '#0a6bef',
      marginTop: '0',
      marginLeft: '4px',
      borderRadius: '10px 10px 10px 0',
      padding: '6px',
      color: 'white',
    },
  },
};

export default styles;
