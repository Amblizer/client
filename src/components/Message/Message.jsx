import React, { PropTypes } from 'react';
import { connect } from 'react-redux';

import './Message.css';

const CHAT_OPERATOR = 'OPERATOR';

export const MessageComponent = (props) => {
  const { chatStyle, author, timestamp, key } = props;
  const content = props.content.map((message, index) => <li key={index}>{message}</li>);

  let message = (
    <div>
      <ul className={`Message-userContent_${chatStyle}`}>
        {content}
      </ul>
    </div>
  );

  if (author === CHAT_OPERATOR) {
    message = (
      <div>
        <div className={`Message-operatorPicture_${chatStyle}`}>
          <img alt="Operator" className={`Message-operatorPictureImage_${chatStyle}`} src="http://placehold.it/40x40/" />
        </div>
        <ul className={`Message-operatorContent_${chatStyle}`}>
          {content}
        </ul>
      </div>
    );
  }

  return (
    <li id={`message_${key}`} className={`Message-message_${chatStyle}`} style={{ clear: 'both' }}>
      {message}
      <span className={`Message-message-timestamp_${chatStyle}`}>{timestamp}</span>
    </li>
  );
};

MessageComponent.propTypes = {
  chatStyle: PropTypes.string,
  timestamp: PropTypes.number,
  key: PropTypes.string,
  author: PropTypes.string,
  content: PropTypes.arrayOf(
    PropTypes.string,
  ),
};

const mapStateToProps = state => ({
  chatStyle: state.ui.chatStyle,
});

const mapDispatchToProps = dispatch => ({
  dispatch,
});


const Message = connect(
  mapStateToProps,
  mapDispatchToProps,
)(MessageComponent);

export default Message;
