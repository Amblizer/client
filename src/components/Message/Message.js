import React, {PropTypes} from 'react';
import injectSheet from 'react-jss';
import { connect } from 'react-redux';
import './Message_styles.css';

const CHAT_OPERATOR = 'OPERATOR'

const Message = props => {
  const content = props.content.map((message, index) => <li key={index}>{message}</li>);
  console.log('props for MESSAGE, ', props);

  let message = (
    <div>
      <ul className="Message-userContent">
        {content}
      </ul>
    </div>
  );

  if (props.author === CHAT_OPERATOR) {
    message = (
      <div>
        <div className="Message-operatorPicture">
          <img alt="Operator" className="Message-operatorPictureImage" src="http://placehold.it/40x40/" />
        </div>
        <ul className="Message-operatorContent">
          {content}
        </ul>
      </div>
    );
  }

  return (
    <li id={`message_${props.id}`} className="letschat-message" style={{ clear: 'both' }}>
      {message}
      <span className="letschat-message-timestamp">{props.timestamp}</span>
    </li>
  );
};

Message.propTypes = {
  author: PropTypes.string,
  content: PropTypes.array,
};

export default Message;