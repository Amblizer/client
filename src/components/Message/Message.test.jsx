import React from 'react';

// import { describe, it, expect } from 'jest';
import { shallow } from 'enzyme';

import Message from './Message.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({
    ui: { chatStyle: 'TEST' },
  })),
};

describe('Message', () => {
  it('matches snapshot', () => {
    const message = {
      key: 0,
      author: 'Test User',
      content: ['Foo Bar'],
    };

    const component = shallow(
      <Message
        store={store}
        key={message.key}
        author={message.author}
        content={message.content}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
