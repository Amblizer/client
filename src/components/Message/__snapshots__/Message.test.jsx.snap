exports[`Message matches snapshot 1`] = `
<Component
  author="Test User"
  chatStyle="TEST"
  content={
    Array [
      "Foo Bar",
    ]
  }
  dispatch={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
